### DISCUSSION TITLE
Is '\' Legal in a Terminal Rule?

### DISCUSSION THREAD

**Discussion by ballcoach12** [eclipse-langium/langium#884] [General]
The following rule generates an error in the grammar editor:

```
terminal S_BYTE_CHAR_STR returns string:
 	'\'' S_BYTE_CHAR_VALUE* '\''
 ;
```

The second backslash causes the following error: `Unexpected character value: ->\<-`.   Do I need to escape this character somehow?

**Comment by msujew** [comment]
Hey @ballcoach12,

I believe our internal string terminal rule doesn't allow escaping. You can either use `"'"` or the corresponding regex (which it will get transformed into anyway) `/'/`.

**Comment by spoenemann** [comment]
We simply don't support escaping characters in strings yet. I created #885 for this.
